pkg/controller/replicator.go

```diff
@@ -76,6 +76,10 @@ func (r *replicator) RemoveBackend(address string) {
 
 	logrus.Infof("Removing backend: %s", address)
 
+	// Stop the monitoring goroutine in the Controller
+	monitorChan := backend.backend.GetMonitorChannel()
```

Comment:
The comment below mentions that backend can be nil, then this would throw a nil pointer.
Did you investigate whether it's safe to rely on this not being nil?

Reply:
I moved it into the if.